internal/pkg/overlay/overlay.go

```diff
@@ -238,12 +238,17 @@ func BuildOverlayIndir(nodeData node.NodeConf, overlayNames []string, outputDir
 					// search for magic file name comment
 					fileScanner := bufio.NewScanner(bytes.NewReader(buffer.Bytes()))
 					fileScanner.Split(ScanLines)
-					reg := regexp.MustCompile(`.*{{\s*/\*\s*file\s*["'](.*)["']\s*\*/\s*}}.*`)
+					regFile := regexp.MustCompile(`.*{{\s*/\*\s*file\s*["'](.*)["']\s*\*/\s*}}.*`)
+					regLink := regexp.MustCompile(`.*{{\s*/\*\s*softlink\s*["'](.*)["']\s*\*/\s*}}.*`)
 					foundFileComment := false
 					for fileScanner.Scan() {
 						line := fileScanner.Text()
-						filenameFromTemplate := reg.FindAllStringSubmatch(line, -1)
-						if len(filenameFromTemplate) != 0 {
+						filenameFromTemplate := regFile.FindAllStringSubmatch(line, -1)
+						softlinkFromTemplate := regLink.FindAllStringSubmatch(line, -1)
+						if len(softlinkFromTemplate) != 0 {
+							wwlog.Debug("Creating soft link %s -> %s", destFileName, softlinkFromTemplate[0][1])
+							return os.Symlink(softlinkFromTemplate[0][1], path.Join(outputDir, destFileName))
+						} else if len(filenameFromTemplate) != 0 {
```

Comment:
This implicitly uses the symlink value if it exists, ignoring any `file` directive. I think specifying both `symlink` and `file` should be an error. Frankly, if there's a `symlink` directive and there's anything else generated that _isn't_ the `symlink` directive, I think that should probably be an error.